### PR TITLE
feat: enhance non-fatals support

### PR DESCRIPTION
**Pull request by ahmedAlaaInstabug**
## Description of the change
1. add non-fatal API
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## Related issues

JIRA ID: MOB-13793

### Development
- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
### Code review 
- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: android/src/main/java/com/instabug/flutter/modules/CrashReportingApi.java [hunk 1]
```diff
@@ -46,4 +53,26 @@ public void send(@NonNull String jsonCrash, @NonNull Boolean isHandled) {
...
+    public void sendNonFatalError(@NonNull String jsonCrash, @Nullable Map<String, String> userAttributes, @Nullable String fingerprint, @NonNull String nonFatalExceptionLevel) {
+        try {
+            Method method = Reflection.getMethod(Class.forName("com.instabug.crash.CrashReporting"), "reportException", JSONObject.class, boolean.class,
+                    Map.class, JSONObject.class, IBGNonFatalException.Level.class);
+            final JSONObject exceptionObject = new JSONObject(jsonCrash);
+
+            JSONObject fingerprintObj = null;
+            if (fingerprint != null) {
+                fingerprintObj = getFingerprintObject(fingerprint);
+//                Method getFingerPrintMethod = Reflection.getMethod(Class.forName("com.instabug.crash.CrashReporting"), "getFingerprintObject", String.class);
```
Comment (AndrewAminInstabug) [inline android/src/main/java/com/instabug/flutter/modules/CrashReportingApi.java:66]: please remove hashed code

File: example/ios/InstabugTests/CrashReportingApiTests.m [hunk 2]
```diff
@@ -36,8 +37,26 @@ - (void)testSend {
...
+    
+    FlutterError *error;
+    
+    [self.api sendNonFatalErrorJsonCrash:jsonCrash userAttributes:userAttributes fingerprint:fingerPrint nonFatalExceptionLevel:ibgNonFatalLevel error:&error];
+    
+    OCMVerify([self.mCrashReporting cp_reportNonFatalCrashWithStackTrace:@{}
+           level:IBGNonFatalLevelError
+         groupingString:fingerPrint
+        userAttributes:userAttributes
+              ]);
```
Comment (a7medev) [inline example/ios/InstabugTests/CrashReportingApiTests.m:59]: Just a little bit of formatting for Objective-C 😅

```suggestion
    NSString *fingerprint = @"fingerprint";
    NSDictionary *userAttributes = @{ @"key" : @"value" };
    NSString *level = @"NonFatalExceptionLevel.error";
    
    FlutterError *error;
    
    [self.api sendNonFatalErrorJsonCrash:jsonCrash
                          userAttributes:userAttributes
                             fingerprint:fingerprint
                  nonFatalExceptionLevel:level
                                   error:&error];
    
    OCMVerify([self.mCrashReporting cp_reportNonFatalCrashWithStackTrace:@{}
                                                                   level:IBGNonFatalLevelError
                                                          groupingString:fingerprint
                                                          userAttributes:userAttributes]);
```

File: ios/Classes/Modules/CrashReportingApi.m [hunk 3]
```diff
@@ -19,10 +21,28 @@ - (void)sendJsonCrash:(NSString *)jsonCrash isHandled:(NSNumber *)isHandled erro
...
-        [[Instabug class] performSelector:reportCrashWithStackTraceSEL withObject:stackTrace withObject:isHandled];
+    NSInteger value = [isHandled integerValue];
+
+    if (value == 1) {
+        [IBGCrashReporting cp_reportNonFatalCrashWithStackTrace:stackTrace level:IBGNonFatalLevelError groupingString:nil userAttributes:nil
+        ];
+    } else {
+        [IBGCrashReporting cp_reportFatalCrashWithStackTrace:stackTrace  ];
+    
     }
```
Comment (a7medev) [inline ios/Classes/Modules/CrashReportingApi.m:32]: We can better cast `isHandled` to `BOOL` and rename the `value` variable to something like `isNonFatal` for clarity, and a little bit of formatting. What do you think?

```suggestion
    BOOL isNonFatal = [isHandled boolValue];
    if (isNonFatal) {
        [IBGCrashReporting cp_reportNonFatalCrashWithStackTrace:stackTrace
                                                          level:IBGNonFatalLevelError
                                                 groupingString:nil
                                                 userAttributes:nil];
    } else {
        [IBGCrashReporting cp_reportFatalCrashWithStackTrace:stackTrace];
    }
```

**Review by a7medev — CHANGES_REQUESTED** [review]
Good job @ahmedAlaaInstabug! 👏🏼 I have added some comments so please check them and let me know your thoughts. 🙏🏼

Side note: some iOS code/tests are broken because the snapshot is on `v12.7.1` which doesn't contain App Flows and this branch's base has App Flows in it.

**Review by a7medev — DISMISSED** [review]
Perfetto! 👌🏼

**Review by AndrewAminInstabug — APPROVED** [review]

**Pull request merged**